airflow/providers/google/cloud/triggers/pubsub.py

```diff
@@ -106,7 +106,10 @@ async def run(self) -> AsyncIterator[TriggerEvent]:  # type: ignore[override]
                 ):
                     if self.ack_messages:
                         await self.message_acknowledgement(pulled_messages)
-                    yield TriggerEvent({"status": "success", "message": pulled_messages})
+
+                    messages_json = [ReceivedMessage.to_dict(m) for m in pulled_messages]
```

Comment:
@gopidesupavan I see the reason why we need this conversion, but could you please move this code in `execute_complete` here: https://github.com/apache/airflow/blob/main/airflow/providers/google/cloud/sensors/pubsub.py#L176 ?
In my opinion `execute_complete` is the better place to do this conversion.

Reply:
Agree , but as per triggers design the trigger event format converts to string https://github.com/apache/airflow/blob/main/airflow/triggers/base.py#L138

We loose the access to original pubsub response structure, when we receive the event in execute complete. its coming as string format. the traiged issue is about this one. Below format we receive in execute complete event. because the trigger event nature. 

So to get the proper response back to execute_complete, we have to do conversion inside the triggerer. 

Current execute_complete:

"[ack_id: 'RFAGFixdRkhRNxkIaFEOT14jPzUgKEUSCQdPAihdeTFTLUFdfWhRDRlyfWB8a1MbBgNGBysOURsHaE5tdR_L0ZL0S0NUa1gSBwVCVnldUhwPbF1ZdQN58b3b8qzgnn8JOjrfj_XZbTuLvKsbZiM9XhJLLD5-LzlFQV5AEkwkDERJUytDCypYEU4EISE-MD5FUw' message { data: 'hello from cloud console!' message_id: '12167757297241504' publish_time { seconds: 1724839443 nanos: 948000000 } } ]"

Reply:
Hope i am make sense :)

Reply:
@gopidesupavan you are right, but now I don't understand what you want to achieve with this PR. Because in an issue https://github.com/apache/airflow/issues/41877 I see that the user wants to see the same XCom result on deff and non-deff mode. If you add conversion in trigger side then trigger returns string to `execute_complete` and after that a `return` statement adds this value to XCom. As a result we will still have different values on deff and non-deff mode.

Also, I see that in non-deff mode the result value depends on `messages_callback` function https://github.com/apache/airflow/blob/main/airflow/providers/google/cloud/sensors/pubsub.py#L146C1-L150C1
In this case for us it means that for consistent values we need to use `handle_messages` function for conversion in deff mode either.

Reply:
yes agree, if user provides `messages_callback` it still produces differences. 

if the user doesn't provide any `messages_callback` there is default conversion sensors uses https://github.com/apache/airflow/blob/main/airflow/providers/google/cloud/sensors/pubsub.py#L189, so
Am thinking atleast the default message response should sync in deff/non deff mode,  IMHO if user uses deferrable mode the message can not be used as it is completely unformatted.

 I can add a note to the doc without `messages_callback` it produces same result to xcom something like this?

Reply:
Another option i would think, but this add extra conversion,

convert to json string format(in trigger)-->and in execute_complete convert back to ReceiveMessage format(it is original pubsub response structure)---> and use user provided message_callback?, with this way we can support messages_callback in both modes.

Reply:
> Another option i would think, but this add extra conversion,
> 
> convert to json string format(in trigger)-->and in execute_complete convert back to ReceiveMessage format(it is original pubsub response structure)---> and use user provided message_callback?, with this way we can support messages_callback in both modes.

@gopidesupavan I like this idea

Reply:
Thanks, i have updated it now to as per above approach.